### PR TITLE
MGMT-23849: fixes unsafe CEL interpolation in describe computeinstance

### PR DESCRIPTION
**Pull request by wgordon17**
## Summary
- Replaces unsafe '%s' CEL interpolation with %q format verb to prevent injection

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of special characters in filter expressions to ensure proper escaping of quotes and backslashes.

* **Tests**
  * Added test coverage for edge cases in filter translation including special characters and escape sequence handling.

### CODE REVIEW

File: internal/cmd/cli/describe/computeinstance/describe_computeinstance_test.go [hunk 1]
```diff
@@ -99,3 +106,138 @@ var _ = Describe("Describe Compute Instance", func() {
...
+		Expect(err).ToNot(HaveOccurred())
+
+		client := publicv1.NewComputeInstancesClient(conn)
+		err = describeComputeInstance(ctx, client, cons, "ambiguous-name")
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("multiple compute instances match"))
+	})
+})
+
+var _ = Describe("CEL filter construction", func() {
```
Comment (adriengentil) [inline internal/cmd/cli/describe/computeinstance/describe_computeinstance_test.go]: I think we can remove, it's doesn't really test any of the code in this project

**Review by adriengentil — APPROVED** [review]

**Review by jhernand — APPROVED** [review]
